ios/Events.m

```diff
@@ -43,16 +43,16 @@ - (void)emitEventInternal:(NSNotification *)notification
 {
   NSArray *eventDetails = [notification.userInfo valueForKey:@"detail"];
   NSString *eventName = [eventDetails objectAtIndex:0];
-  NSDictionary *eventData = [eventDetails objectAtIndex:1];
-
+  id eventData;
+  if (eventDetails.count > 1) {
+    eventData = [eventDetails objectAtIndex:1];
+  }
   [self sendEventWithName:eventName body:eventData];
 }
 
-+ (void)emitEventWithName:(NSString *)name andPayload:(NSString *)payload
++ (void)emitEventWithName:(NSString *)name andPayload:(NSString *)base64string
 {
-  NSData *data = [payload dataUsingEncoding:NSUTF8StringEncoding];
-  NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
-  NSDictionary *eventDetail = @{@"detail":@[name,json]};
+  NSDictionary *eventDetail = @{@"detail": base64string ? @[name, base64string] : @[name]};
   [[NSNotificationCenter defaultCenter] postNotificationName:@"event-emitted"
```

Comment:
Anyone know why we do this loop through `NSNotificationCenter` when it seems we could just call `[self sendEventWithName:eventName body:eventData]` right here?

Reply:
I basically copy pasted all that from https://stackoverflow.com/questions/38818287/react-native-bridge-is-nil-when-i-call-method-from-another-method#answer-43543711 (link at top of file). Maybe something in there.